salvo/src/lib/test_source_tree.py

```diff
@@ -9,135 +9,428 @@
 from src.lib import (cmd_exec, source_tree)
 import api.source_pb2 as proto_source
 
+logging.basicConfig(level=logging.DEBUG)
 log = logging.getLogger(__name__)
 
+_GIT_TAG_LIST = """
+v1.14.5
```

Comment:
Thank you for making the list shorter. This is still bit longer than I expected, indicating that I might be missing something about the test cases. Following the [Zero one infinity rule](https://en.wikipedia.org/wiki/Zero_one_infinity_rule), do we need more than one or two tags to test the behavior we are after? Any more than the bare minimum makes it unclear as to what the tests depend on / really need.

Reply:
I wanted more than 2 tags in the list to verify we can discover an arbitrary number of commits backwards from a given hash.  Originally the intent was that we would test the hash N and N-1, but it maybe useful to test N-3 etc.  Having more than 2 tags in the list gives us the ability to verify this.

That said, I shortened the lists where applicable.